### PR TITLE
Normalize device to CPU when evaluating

### PR DESCRIPTION
**Pull request by tomaarsen**
Based on discussion in https://github.com/huggingface/setfit/issues/339#issuecomment-1488142771

Hello!

## Pull Request overview
* Ensure that `y_pred` is always on the CPU when using a differentiable head.
* Test to ensure that `y_pred` is on CPU.

## Details
As discussed in #339, you can get odd errors in your custom evaluation functions when switching to using a GPU or not. This PR simplifies that a bit by always putting a torch `y_pred` on the CPU.
I believe that Transformers goes one step further, and it calls `y_pred.cpu().numpy()` (perhaps with a `.detach()` too). To me, that would also be a reasonable alternative. However, for now I've just moved it to the CPU.

- Tom Aarsen

### CODE REVIEW

File: src/setfit/trainer.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import evaluate
 import numpy as np
+import torch
```
Comment (alvarobartt) [inline src/setfit/trainer.py:6]: Hi @tomaarsen, maybe you could place the `torch` import inside the `TYPE_CHECKING` block as described my following comment
